tests/regression/anomaly/test_anomaly_classificaiton.py

```diff
@@ -0,0 +1,240 @@
+"""Tests for Anomaly Classification with OTX CLI."""
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+import json
+import os
+import random
+from pathlib import Path
+from timeit import default_timer as timer
+
+import pytest
+
+from otx.cli.registry import Registry
+from tests.regression.regression_test_helpers import (
+    ANOMALY_DATASET_CATEGORIES,
+    TIME_LOG,
+    get_result_dict,
+    get_template_performance,
+    load_regression_configuration,
+)
+from tests.test_suite.e2e_test_system import e2e_pytest_component
+from tests.test_suite.run_test_command import (
+    nncf_eval_testing,
+    nncf_optimize_testing,
+    otx_deploy_openvino_testing,
+    otx_eval_compare,
+    otx_eval_deployment_testing,
+    otx_eval_e2e_eval_time,
+    otx_eval_e2e_train_time,
+    otx_eval_openvino_testing,
+    otx_export_testing,
+    otx_train_testing,
+    pot_eval_testing,
+    pot_optimize_testing,
+)
+
+# Configurations for regression test.
+TASK_TYPE = "anomaly_classification"
+SAMPLED_ANOMALY_DATASET_CATEGORIES = random.sample(ANOMALY_DATASET_CATEGORIES, 1)
+
+otx_dir = os.getcwd()
+templates = Registry("otx/algorithms/anomaly").filter(task_type=TASK_TYPE.upper()).templates
+templates_ids = [template.model_template_id for template in templates]
+
+result_dict = get_result_dict(TASK_TYPE)
+result_dir = f"/tmp/regression_test_results/{TASK_TYPE}"
```

Comment:
Personally, I would prefer fixtures here as they will ensure that the temporary directories are cleaned even if the test fails or is interrupted. Since we don't restart the CI is possible that tmp is not cleared. In this case we assume that overwrite will solve the issue.

```python
@pytest.fixture(scope="class")
def result_dir():
    with TemporaryDirectory(f"regression_test_results/{TASK_TYPE}") as tmpdir:
        yield tmpdir
```

> _Note:_ This is an example out of my head so I haven't tested it. This snippet might require some modification.

Other option is this: https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html#the-tmp-path-fixture

Reply:
For now, we want to keep the regression test results. If we can make automatic csv report to CI, we can change this part as you suggested